lib/models/LifecycleConfiguration.ts

```diff
@@ -105,15 +117,42 @@ export default class LifecycleConfiguration {
      * @param config - the CloudServer config
      * @return - LifecycleConfiguration instance
      */
-    constructor(xml: any, config: { replicationEndpoints: { site: string }[] }) {
+    constructor(xml: any, config: { replicationEndpoints: { site: string }[], supportedLifecycleRules: string[] }) {
         this._parsedXML = xml;
         this._storageClasses =
             config.replicationEndpoints.map(endpoint => endpoint.site);
+        this._supportedLifecycleRules = LifecycleConfiguration._buildSupportedLifecycleRules(
+            config.supportedLifecycleRules,
+        );
         this._ruleIDs = [];
         this._tagKeys = [];
         this._config = {};
     }
 
+    // Memoize the supported lifecycle rules to avoid recalculating them
```

Comment:
```suggestion
    // Memoize the supported lifecycle rules to avoid recalculating them
```
Memorize the supported lifecycle rules to avoid recalculating them

Reply:
No typo, [Memoization](https://en.wikipedia.org/wiki/Memoization) is the right term here